### PR TITLE
Consistent error logging for cache

### PR DESCRIPTION
**Pull request by kkmuffme**
Generally, psalm throws errors in cache.

However there were 2 places, where error_log was used/leftover from early versions. Changed now for consistency.

### CODE REVIEW

**Comment by kkmuffme** [comment]
@orklah there is a bug in psalm unrelated to this PR? Could you please quickly check

Test fails: https://github.com/vimeo/psalm/runs/7860346355?check_suite_focus=true (which will cause other tests to fail too), due to $config being null (how is this even possible when construct is called with Config, as it should already give an error then??)

When I add a check for that, it says config cannot be null:
https://github.com/vimeo/psalm/runs/7860822542?check_suite_focus=true

**Comment by AndrolGenhald** [comment]
`Psalm\Tests\Internal\Provider\ParserInstanceCacheProvider` extends `ParserCacheProvider` and overrides its constructor to do nothing, so the parent constructor is never called.

**Comment by kkmuffme** [comment]
Ok so I guess I will put back the call to static config as it was there before. (since properly fixing this would mean rewriting tons of code that is unrelated to this PR)

File: src/Psalm/Config.php [hunk 1]
```diff
@@ -1013,8 +1013,16 @@ private static function fromXmlAndPaths(
             chdir($config->base_dir);
         }
 
-        if (is_dir($config->cache_directory) === false && @mkdir($config->cache_directory, 0777, true) === false) {
-            error_log('Could not create cache directory: ' . $config->cache_directory);
+        if (!is_dir($config->cache_directory)) {
+            try {
+                mkdir($config->cache_directory, 0777, true);
```
Comment (orklah) [inline src/Psalm/Config.php:1018]: seems weird to not check mkdir return. Does it throws on any error with some config?

**Comment by orklah** [comment]
Thanks

**Pull request merged**